testing/MLDB-1322-sum_stem_token.py

```diff
@@ -0,0 +1,206 @@
+
+def query(q):
+    return mldb.perform('GET', '/v1/query', [['q', q]])
+
+
+ds = mldb.create_dataset({
+    'type': 'sparse.mutable',
+    'id': 'veggies'})
+ds.record_row('row_0', [['txt', 'potato,carrots', 0], ['label',0, 0]])
+ds.record_row('row_1', [['txt', 'potato,potatoes,potato', 0], ['label', 0, 0]])
+ds.record_row('row_2', [['txt', 'carrot,carrots', 0], ['label',1, 0]])
+
+# We have a dataset that looks like this
+#
+#      txt                 label
+#   potato,carrots           0
+# potato,potatoes,potato     0
+# carrot,carrots             1
+#
+# Ultimately I want to count how many of each veggy per label.
+# Let's do it step by step.
+
+
+# Step 1: tokenize
+q = '''
+SELECT tokenize(txt) as *, label
+FROM veggies
+'''
+res = query(q)
+res = check_res(res, 200)
+
+expected = {
+    'row_0': {
+        'potato': 1,
+        'carrots': 1,
+        'label': 0},
+    'row_1': {
+        'potato': 2,
+        'potatoes': 1,
+        'label': 0},
+    'row_2': {
+        'carrot': 1,
+        'carrots': 1,
+        'label': 1}}
+
+# We now have
+#
+#    potato  potatoes carrot  carrots   label
+#       1       nan     nan     1         0
+#       2        1      nan    nan        0
+#      nan      nan      1      1         1
+
+check_output(res, expected)
+
+# Step 2: tokenize and then stem
+
+conf = {
+    "type": "stemmer",
+    "params": {
+        "language": "english"
+    }
+}
+res = mldb.perform("PUT", "/v1/functions/stem", [], conf)
+
+q = '''
+SELECT stem({words: {tokenize(txt) as *}})[words] as *, label
+FROM veggies
+'''
+res = query(q)
+res = check_res(res, 200)
+
+expected = {
+    'row_0': {
+        'potato': 1,
+        'carrot': 1,
+        'label': 0},
+    'row_1': {
+        'potato': 3,
+        'label': 0},
+    'row_2': {
+        'carrot': 2,
+        'label': 1}}
+# We now have
+#
+#    potato  carrot   label
+#       1       1       0
+#       3      nan      0
+#      nan      2       1
+
+check_output(res, expected)
+
+# Step 3: Let start with just summing the columns
+q = '''
+SELECT sum(stem({words: {tokenize(txt) as *}})[words]) as *
+FROM veggies
+'''
+res = query(q)
+res = check_res(res, 200)
+
+expected = {
+    '[]': {
+        'potato': 4,
+        'carrot': 3}}
+check_output(res, expected)
+
+# Step 4: same but grouped by label
+q = '''
+SELECT sum(stem({words: {tokenize(txt) as *}})[words]) as *, label
+FROM veggies
+GROUP BY label
+'''
+res = query(q)
+res = check_res(res, 200)
+
+expected = {
+    '[0]': {
+        'potato': 4,
+        'carrot': 1,
+        'label': 0},
+    '[1]': {
+        'carrot': 2,
+        'label': 1
+    }}
+check_output(res, expected)
+
+
+# Step 6: same thing but from a merged dataset
+ds = mldb.create_dataset({
+    'type': 'sparse.mutable',
+    'id': 'label2'})
+ds.record_row('row_0', [['label2',2, 0]])
+ds.record_row('row_1', [['label2', 2, 0]])
+ds.record_row('row_2', [['label2',3, 0]])
+
+q = '''
+SELECT sum(stem({words: {tokenize(txt) as *}})[words]) as *
```

Comment:
can you please add a case for `sum(stem({words: tokenize(txt) })[words]) as *`? i.e. without the "{ ... as *}` around the `tokenize()`? If MLDB doesn't support this, please comment out the test case and I will create an issue.